python/mdvtools/auth/authutils.py

```
@@ -0,0 +1,359 @@
+
+    except Exception as e:
+        logger.exception(f"validate_sso_user error: {e}")
+        return None, jsonify({"error": "Internal server error - user not validated"}), 500
+
+
+def maybe_require_user(ENABLE_AUTH):
+    def decorator(f):
+        @wraps(f)
+        def decorated_function(*args, **kwargs):
+            if not ENABLE_AUTH:
+                return f(user=None, *args, **kwargs)  # Inject `user=None` for consistency
```

Ruff (0.8.2)

Star-arg unpacking after a keyword argument is strongly discouraged

(B026)